lib/Cake/Network/CakeSocket.php

```diff
@@ -129,8 +129,12 @@ public function connect() {
 			$this->disconnect();
 		}
 
+		$hasProtocol = strpos($this->config['host'], '://') !== false;
```

Comment:
With `parse_url` the relevant code would look like this:
```php
		$urlParts = parse_url($this->config['host']);
		$this->config['host'] = $urlParts['host'];
		if (isset($urlParts['scheme'])) {
			$this->config['protocol'] = $urlParts['scheme'];
		}
```
Maybe it is subject to taste, but to me this looks cleaner. Thanks for the reminder @AD7six